### ISSUE TITLE
pipe text object is broken on evil-git-head

### ISSUE THREAD

**Issue by ninrod** [noctuid/targets.el#19]
Steps to reproduce. just try to use the `|` text object with the latest evil-mode (from git-head)

```elisp
(use-package targets
    :load-path "~/.dotfiles/deps/emacs/ninrod/targets.el"
    :init
    (setq targets-user-text-objects '((pipe "|" nil separator)
                                      (paren "(" ")" pair :more-keys "b")
                                      (bracket "[" "]" pair :more-keys "r")
                                      (curly "{" "}" pair :more-keys "c")))
    :config
    (targets-setup t
                   :inside-key nil
                   :around-key nil
                   :remote-key nil))
```

interesting to note, on commit 6a3f3c6, `ci|` puts me in insert mode, but is does nothing. 

in commit 65ae10b, `ci|` is corrupted. you can use `vi|` to see that the selection behaves very strangely. 

ps: I use the pipe text object extensively for markdown table editing. That's why I noticed it. 

ps2: the pipe operator was the use case that originated `targets.el` in the first place, right [here](https://github.com/noctuid/evil-textobj-anyblock/issues/9)

ps3: __breakthrough__: ~I know for a fact that it works in emacs-evil/evil@47c4c03 There were 75 more commits since this commit. This should be a nice candidate to a git bisect.~
__UPDATE__: I found the offending commit on evil: 

it's emacs-evil/evil@d457729, which is a merge of commit:

`a5465f3: bisect fix evil up block -- make it symmetrical forward and backward` by @willghatch 

you can verify that it works in the previous commit: emacs-evil/evil@bc753f4:

![image](https://user-images.githubusercontent.com/8352747/28735877-b2441bce-73bd-11e7-9fbf-1029d439ac78.png)

here is the offending merged pull: emacs-evil/evil#747

__NOTE__: emacs-evil/evil@a5465f3 can be reverted cleanly from evil git head. This solves the problem.

ps4: I've also noticed that the macro [discussed](https://github.com/noctuid/evil-textobj-anyblock/issues/9#issuecomment-263761927) in noctuid/evil-textobj-anyblock#9 breaks in evil-git-head. So I opened an issue in evil itself: emacs-evil/evil#884

**Comment by ninrod** [comment]
__update__: @willghatch wrote patch emacs-evil/evil#889 that aims to fix emacs-evil/evil#747

**Comment by ninrod** [comment]
emacs-evil/evil#889 was merged in master.
